### DISCUSSION TITLE
setTimeout and setInterval  will be executed during build

### DISCUSSION THREAD

**Discussion by hahaxiaowai** [vuejs/vitepress#3977] [Q&A]
I use custom component in vitepress, when I use `inteval` ，it will be exceuted during build.
this my code
import
```js
// index.js
import Test from "./Test.vue";

export default {
  extends: theme,
  Layout: () => {
    return h(theme.Layout, null, {

    });
  },
  enhanceApp({ app, router, siteData }) {
    app.component("test", Test);
  },
} 
```
use
``` md
---
layout: test
---
```
Test.vue
```vue
<script setup">
setInterval(()=>{
  console.log('do')
})
</script>

<template>
</template>

<style scoped>

</style>
```
build, and will not stop, why?
![image](https://github.com/vuejs/vitepress/assets/49934348/ebfa4b7b-4f77-4752-a939-f980941f76a4)

**Comment by brc-dd** [comment]
run them inside onMounted hook / inBrowser check if you don't want to run them during SSR, or wrap your component with ClientOnly.
